### ISSUE TITLE
AzCore compile errors with VS2022

### ISSUE THREAD

**Issue by AMZN-Olex** [o3de/o3de#9592]
**Describe the bug**
AzCore with VS2022 17.2.1 results in a compile error:

```
1>C:\git\o3de\Code\Framework\AzCore\AzCore/RTTI/BehaviorContext.h(3038,21): error C2220: the following warning is treated as an error
1>                    AZ_Assert(bc, "behavior class is invalid for typeId: %s", typeId.ToString<AZStd::string>().c_str());
1>                    ^
```

**Steps to reproduce**
1. Configure O3DE for VS2022
2. Build AzCore target

**Expected behavior**
AzCore compiles

**Actual behavior**
AzCore fails to compile

**Found in Branch**
development

**Desktop/Device (please complete the following information):**
 - Device: PC
 - OS: Windows
 - Version 11
 - CPU Ryzen 5900x
 - GPU NVidia RTX 3090
 - Memory 64GB

**Comment by AMZN-Olex** [comment]
The fix seems to be
```
diff --git a/Code/Framework/AzCore/AzCore/Debug/Trace.h b/Code/Framework/AzCore/AzCore/Debug/Trace.h
index fe33bf1b9c..01848990fd 100644
--- a/Code/Framework/AzCore/AzCore/Debug/Trace.h
+++ b/Code/Framework/AzCore/AzCore/Debug/Trace.h
@@ -138,7 +138,7 @@ namespace AZ
     {                                                                                                                                                                                \
         using namespace AZ::TraceInternal;                                                                                                                                           \
         [[maybe_unused]] const auto& rTraceFmtCompileTimeCheckExpressionHelper = (expression); /* This is needed for edge cases for expressions containing lambdas, that ere unsupported before C++20 */   \
-        constexpr ExpressionValidResult isValidTraceFmtResult = ExpressionIsValid<decltype(rTraceFmtCompileTimeCheckExpressionHelper)>::value;                                       \
+        [[maybe_unused]] constexpr ExpressionValidResult isValidTraceFmtResult = ExpressionIsValid<decltype(rTraceFmtCompileTimeCheckExpressionHelper)>::value;                                       \
         /* Assert different message depending whether it's const char array or if we have extra arguments */                                                                         \
         static_assert(!(isVaArgs) ? isValidTraceFmtResult != ExpressionValidResult::Invalid_ConstCharArray : true, baseMsg " " msg);                                                   \
         static_assert(isVaArgs  ? isValidTraceFmtResult != ExpressionValidResult::Invalid_ConstCharArray : true, baseMsg " " msgVargs);                                              \
```

**Comment by LB-ArturZieba** [comment]
Issue verified as fixed on the Stabilization/2210 (f410a0a) build.
